src/main/kotlin/org/jitsi/nlj/transform/node/ToggleablePcapWriter.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright @ 2018 - Present, 8x8 Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.jitsi.nlj.PacketInfo
+import org.jitsi.nlj.transform.node.ObserverNode
+import org.jitsi.nlj.transform.node.PcapWriter
+import org.jitsi.utils.logging2.Logger
+import java.util.Date
+
+class ToggleablePcapWriter(
```

Comment:
Would be good to have a class javadoc here.  Especially with a comment about the (lack of) thread safety.

Reply:
i've followed the stepped into pcap4j and saw some synchronization primitives; do you know the writes to the pcap file need to be synched?